### PR TITLE
CE-2352 Monitoring Oracle DB

### PR DESCRIPTION
**Pull request by Neelakshi1808**

### CODE REVIEW

File: oracle/oracle.rb [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+    ['--debug',     '-d', GetoptLong::NO_ARGUMENT],
+    ['--verbose',   '-v', GetoptLong::NO_ARGUMENT],
+    ['--config',    '-c', GetoptLong::REQUIRED_ARGUMENT],
+    ['--apikey',    '-k', GetoptLong::REQUIRED_ARGUMENT],
+    ['--frequency', '-f', GetoptLong::REQUIRED_ARGUMENT],
+    ['--apihost',   '-a', GetoptLong::REQUIRED_ARGUMENT]
+)
+
+#base_path = '/usr/local/copperegg/ucm-metrics/oracle'
+base_path = '/home/ubuntu/Desktop'
```
Comment (marutinandanpandya) [inline oracle/oracle.rb]: reminder!!

Comment (Neelakshi1808) [inline oracle/oracle.rb]: Fixed.

File: oracle/oracle.rb [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+    end
+    interruptible_sleep @freq
+  end
+end
+
+
+def ensure_oracle_metric_group(metric_group, group_name, group_label)
+  if metric_group.nil? || !metric_group.is_a?(CopperEgg::MetricGroup)
+    log 'Creating oracle metric group'
+    metric_group = CopperEgg::MetricGroup.new(:name => group_name, :label => group_label, :frequency => @freq)
```
Comment (Neelakshi1808) [inline oracle/oracle.rb]: Done

File: oracle/oracle.rb [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+  metric_group.metrics << {type: 'ce_gauge_f', name: 'connection_time', position: 14,
+                           label: 'Connection Time', unit: 'seconds'}
+  metric_group.save
+  return metric_group
+end
+
+def create_oracle_dashboard(metric_group, name)
+  log 'Creating new Oracle Dashboard'
+  metrics = metric_group.metrics || []
+  CopperEgg::CustomDashboard.create(metric_group, :name => name, :identifiers => nil, :metrics => metrics)
```
Comment (Neelakshi1808) [inline oracle/oracle.rb]: Done

File: oracle/oracle.rb [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+end
+
+####################################################################
+
+# get options
```
Comment (Neelakshi1808) [inline oracle/oracle.rb]: Done.

File: oracle/oracle.rb [hunk 5]
```diff
@@ -0,0 +1,365 @@
...
+                                                  @config[service]['group_label'])
+      end
+
+      raise "Could not create a metric group for #{service}" if metric_group.nil?
+      log "Checking for existence of #{@config[service]['dashboard']}"
+
+      dashboard = dashboards.detect { |d| d.name == @config[service]['dashboard'] } unless dashboards.nil?
+      dashboard = create_oracle_dashboard(metric_group, @config[service]['dashboard']) if dashboard.nil?
+
+      raise "Could not create a dashboard for #{service}" if dashboard.nil?
```
Comment (Neelakshi1808) [inline oracle/oracle.rb]: Fixed.

File: oracle/oracle_installer.sh [hunk 6]
```diff
@@ -0,0 +1,524 @@
...
+
+    echo "Testing with command: sqlplus  -L $USER_NAME/$PASSWORD@(DESCRIPTION=(ADDRESS=(PROTOCOL=TCP)(Host=$URL)(Port=$PORT))) < /dev/null"
+    if [ -n "$ORACLE_HOME" ]; then
+        $ORACLE_HOME/bin/sqlplus  -L "$USER_NAME/$PASSWORD@(DESCRIPTION=(ADDRESS=(PROTOCOL=TCP)(Host=$URL)(Port=$PORT)))" < /dev/null > /tmp/oracle_stats.txt
+    elif [ -n "$LD_LIBRARY_PATH" ]; then
+        $LD_LIBRARY_PATH/sqlplus  -L "$USER_NAME/$PASSWORD@(DESCRIPTION=(ADDRESS=(PROTOCOL=TCP)(Host=$URL)(Port=$PORT)))" < /dev/null > /tmp/oracle_stats.txt
+    else
+        echo "Exiting from script. Cannot access sqlplus to test connection."
+        echo "Ensure ORACLE_HOME or LD_LIBRARY_PATH variables point to correct path for sqlplus"
+        # exit 1
```
Comment (Neelakshi1808) [inline oracle/oracle_installer.sh]: Uncommented the line.

**Pull request merged**